experimental/dds/tree2/src/feature-libraries/default-schema/defaultChangeFamily.ts

```diff
@@ -168,15 +180,36 @@ export class DefaultEditBuilder implements ChangeFamilyEditor, IDefaultEditBuild
 		return {
 			set: (newContent: ITreeCursor | undefined, wasEmpty: boolean): void => {
 				const id = this.modularBuilder.generateId();
-				const optionalChange =
-					newContent === undefined
-						? optional.changeHandler.editor.clear(wasEmpty, id)
-						: optional.changeHandler.editor.set(newContent, wasEmpty, {
-								fill: this.modularBuilder.generateId(),
-								detach: this.modularBuilder.generateId(),
-						  });
+				const edits: EditDescription[] = [];
+				let optionalChange: OptionalChangeset;
+				if (newContent !== undefined) {
+					const fillId = this.modularBuilder.generateId();
+					const detachId = this.modularBuilder.generateId();
+
+					const build = this.modularBuilder.buildTrees(fillId, [newContent]);
+					edits.push(build);
+
+					optionalChange = optional.changeHandler.editor.set(newContent, wasEmpty, {
+						fill: fillId,
+						detach: detachId,
+					});
+				} else {
+					optionalChange = optional.changeHandler.editor.clear(wasEmpty, id);
+				}
+
 				const change: FieldChangeset = brand(optionalChange);
-				this.modularBuilder.submitChange(field, optional.identifier, change);
+				const edit: FieldEditDescription = {
+					type: "field",
+					field,
+					fieldKind: optional.identifier,
+					change,
+				};
+				edits.push(edit);
+				if (newContent !== undefined) {
+				}
+
+				// TODO: What's the deal with this maxId
+				this.modularBuilder.submitChanges(edits);
```

Comment:
You need to pass in the highest ID that was generated. In the case of `newContent !== undefined` that's `detachId`, otherwise that's `id` (based on the current factoring).